sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/xml/XmlInferSchema.scala

```diff
@@ -243,6 +244,22 @@ class XmlInferSchema(options: XmlOptions, caseSensitive: Boolean)
       }
     }
 
+    @tailrec
+    def inferAndCheckEndElement(parser: XMLEventReader): Boolean = {
+      parser.peek match {
+        case _: EndElement | _: EndDocument => true
+        case _: StartElement => false
+        case c: Characters if !c.isWhiteSpace =>
+          val characterType = inferFrom(c.getData)
+          parser.nextEvent()
+          addOrUpdateType(options.valueTag, characterType)
```

Comment:
Is there a test case for this scenario?

Reply:
A valueTag that locates after a closing tag in the inner element and before the closing tag in the outer element will cover this scenario.
```
<a>
    value2
    <b>1</b>
    value3
</a>
```
We covered this case in the most our test cases